scheduler/server/job_state.go

```diff
@@ -197,3 +228,41 @@ func (j *jobState) getJobStatus() domain.Status {
 	}
 	return domain.InProgress
 }
+
+func (j *jobState) addTaskToStartTimeMap(jobClass string, task *taskState, startTimeSec time.Time) {
+	if j.tasksByJobClassAndStartTimeSec == nil {
+		return
```

Comment:
should we log something here?

Reply:
this would only happen if/when we use a different scheduling algorithm - so if we revert to the orig scheduling algorithm, this logging will spam the logs.